packages/block-library/src/heading/index.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Appending the wp-block-heading to before rendering the stored `core/heading` block contents.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Adds a wp-block-heading class to the heading block content.
+ *
+ * For example, the following block content:
+ *  <h2 class="align-left">Hello World</h2>
+ *
+ * Would be transformed to:
+ *  <h2 class="align-left wp-block-heading">Hello World</h2>
+ *
+ * @param array  $attributes Attributes of the block being rendered.
+ * @param string $content Content of the block being rendered.
+ *
+ * @return string The content of the block being rendered.
+ */
+function block_core_heading_render( $attributes, $content ) {
+	if ( ! $content ) {
+		return $content;
+	}
+	// $pattern matches the opening tag of the heading element .
+	$pattern = '/
+        ^(\s*)                               # Any leading whitespace.
+        <(?<tag_name>
+            h[1-6]                           # The opening tag...
+            (?=\s|>)                         # ...followed by a whitespace or >
+                                             # ?= means a "lookahead"
+        )
+        (?<before_class>                     # Any attributes prior to "class"
+            [^>]*?                           # Non-greedy match of any character except ">"
+        )
+        (?:\s+                               # Until we find the class attribute, if any
+            class=(?P<quote>[\'"])           # The quote character
+            (?<class_name>.*?)               # Non-greedy match of any character
+            (\k{quote})                      # Until we find that quote character again
+            (?<after_class>[^>]*?)           # The rest of the tag
+        )?
+        >                                    # The closing tag
+    /xm';
+
+	return preg_replace_callback(
+		$pattern,
+		function ( $matches ) {
+			// Parse the existing class names.
+			$current_class_attr = ! empty( $matches['class_name'] ) ? $matches['class_name'] : '';
```

Comment:
I believe that this line is wholly redundant, because if there's nothing inside the `class_name` match the value of `$matches['class_name']` will be `''`, so we can just use it as-is.